src/Database/LSMTree/Internal/Chunk.hs

```diff
@@ -0,0 +1,98 @@
+module Database.LSMTree.Internal.Chunk
+(
+    -- * Chunks
+    Chunk (Chunk),
+
+    -- * Balers
+    Baler,
+    createBaler,
+    feedBaler,
+    readBalerRemnant
+)
+where
+
+import           Prelude hiding (concat, length, splitAt)
+
+import           Control.Monad.ST.Strict (ST)
+import           Data.STRef.Strict (STRef, newSTRef, readSTRef, writeSTRef)
+import           Data.Vector.Primitive (Vector, concat, length, slice, splitAt)
+import           Data.Word (Word8)
+
+-- * Chunks
+
+-- | A chunk of bytes, typically output during incremental index serialisation.
+newtype Chunk = Chunk (Vector Word8)
+
+-- * Balers
+
+-- Mutable state of a baler.
+data BalerState = BalerState
+                      ![Vector Word8] -- Reverse list of queued blocks
```

Comment:
```suggestion
                      [Vector Word8] -- Reverse list of queued blocks
```

Reply:
Now it was on me to add the `!`. 😉 My motivation was that making this field strict is sensible already from a purely denotational standpoint (so not even considering evaluation, unboxing, and such): this field becoming ⊥ wouldn’t make sense. Actually, it would be even more sensible to make the whole list strict, but this would unnecessarily complicate the code. What would be your motivation for removing the `!`?

Reply:
The change you suggested is in d0384bfa5bf54d4d1debea8ca52098e3a48e5c7b.